app/controllers/initiatives_controller.rb

```diff
@@ -40,19 +40,24 @@ def destroy
       redirect_to root_path
     end
   end
-
+  
   private
-
+  
   def initiative_params
     params.require(:initiative).permit(:title, :description, :target_date, :completion)
   end
-
+  
   def find_initiative
-    @initiative = Initiative.find(params[:id])
+    @initiative = current_user.initiatives.find_by_id(params[:id])
+    
+    if @initiative.nil?
+      flash[:notice] = "You entered wrong routes"
```

Comment:
This notice will be shown to the user, but the user probably doesn't know what routes are--that's internal to Rails. Is there a more user-friendly message we could show here to improve the user experience?

Reply:
modified flash message for invalid route